### PR TITLE
Align transport and docs with RFC audit

### PR DESCRIPTION
**Pull request by aistar-au**
Summary

This change aligns the current HTTP, SSE, TLS, retry, and documentation surface with the RFC audit, tightens the shared stream parser around valid SSE framing, and fixes the two requested regressions in the editor and JWT module docs. It also makes the local SSE endpoints explicitly non-cacheable and non-buffered so clients and intermediaries see the stream the way the server actually emits it.

Motivation

The attached RFC audit identified several places where the implementation and documentation had drifted from the intended HTTP and SSE contract: the client still tolerated non-SSE raw JSON chunks, the local server emitted misleading SSE ids, Retry-After handling only covered numeric values, and the docs did not clearly state the standards posture or the deliberate non-browser POST streaming exception. The request also called out two specific follow-ups: narrow the JWT RFC claims to the HS256 subset that is actually implemented, and restore expected Ctrl+D delete behavior when the input buffer is non-empty.

Approach

The parser was tightened first because it is the contract boundary that other transport fixes depend on. It now strips a UTF-8 BOM once, accepts event, data, id, and retry fields, normalizes \n, \r\n, and \r, strips exactly one leading space per SSE field rules, and stops accepting raw JSON frames without SSE framing. Keeping the raw JSON fallback would have preserved a non-compliant path and hidden malformed upstream streams, so it was removed instead of being documented as supported behavior.

On the request and server side, the client now sends a versioned User-Agent, explicit timeouts, Accept-Encoding: identity, and request-side Cache-Control: no-store, while the local SSE handlers now emit Cache-Control: no-cache, no-store, must-revalidate plus X-Accel-Buffering: no. Timestamp-based SSE ids were removed because they are not stable resume tokens. Retry-After parsing now accepts HTTP-date forms as well as delta-seconds. The remaining editor and JWT follow-ups were kept small and direct: Ctrl+D deletes when text is present and only quits on an empty buffer, and the JWT docs now explicitly describe the implemented HS256/JWS subset instead of implying broader algorithm support.

Validation

- cargo fmt --check
- cargo clippy --all-targets -- -D warnings
- cargo nextest run -j 2
- bash scripts/check_forbidden_names.sh
- cargo run --bin vex -- --help >/dev/null
- mdbook build docs

All of the commands above completed successfully in the sandbox checkout after the parser repair and the follow-up test fixture update for the new client field.

Risks

- Duplication / missed shared-code opportunity: the SSE cache and proxy-buffering headers are now set in both streaming handlers. That duplication is small and explicit, but it could be collapsed into a shared SSE-response helper in a follow-up if more streaming endpoints are added.
- Unused code: no orphaned branches or unused helpers were left behind, but the parser now stores id and retry state without exposing resumable replay yet. That state is intentional groundwork and should either be surfaced by a future resume API or removed if replay is not pursued.
- Bugs / regression risk: the parser now rejects raw JSON streams that are not framed as SSE. That is the intended contract, but any backend that was depending on the old permissive fallback will now fail fast instead of being silently accepted. Retry-After HTTP-date handling also depends on local clock accuracy, so skewed clocks can shorten or lengthen the interpreted backoff.
- Inconsistency with ADRs: no ADR conflict was identified against the active roadmap. The change stays within the existing facade and transport boundaries from ADR-028, preserves the normalized streaming event contract expected by ADR-030 and ADR-031, and only tightens behavior around already-documented transport surfaces rather than introducing a parallel path.
- Test coverage gaps: unit and integration coverage were added for the parser edge cases, Retry-After HTTP-date parsing, SSE response headers, and the Ctrl+D regression, but there is still no live end-to-end backend test that exercises BOM-prefixed frames, id or retry fields, or intermediary buffering behavior against a real upstream server.
- Follow-up work deferred: resumable SSE replay is still out of scope, so event ids remain intentionally absent from local server output. RS256 and ES256 support also remain out of scope and would need an API extension rather than wrapper changes.

### CODE REVIEW

File: src/runtime/rate_limit.rs [hunk 1]
```diff
@@ -59,14 +60,23 @@ fn re_retry_body() -> &'static regex_lite::Regex {
...
+            source: RetryHintSource::Header,
+        });
+    }
+
+    let when = httpdate::parse_http_date(trimmed).ok()?;
+    let duration = when.duration_since(SystemTime::now()).ok()?;
     Some(RetryHint {
-        delay_ms: (seconds * 1000.0) as u64,
+        delay_ms: duration.as_millis().min(u64::MAX as u128) as u64,
         source: RetryHintSource::Header,
```
[the remark this answers is not in the document]
Comment (aistar-au) [reply src/runtime/rate_limit.rs:92]: Addressed in 2f5d86a: past HTTP-date Retry-After values now clamp to 0ms via Duration::ZERO, and coverage was added for a past-date header.

File: src/api/stream.rs [hunk 2]
```diff
@@ -143,38 +145,40 @@ impl StreamParser {
...
         }
 
-        let json_data = if !data_lines.is_empty() {
-            data_lines.join("\n")
-        } else {
-            frame_text.trim().to_string()
-        };
+        if data_lines.is_empty() {
+            return Ok(Vec::new());
+        }
```
[the remark this answers is not in the document]
Comment (aistar-au) [reply src/api/stream.rs:177]: Addressed in 2f5d86a: non-empty raw JSON payload frames without a data: field now emit sse_parse_error instead of being silently ignored, and the regression is covered in src/api/stream/tests.rs.

File: src/api/stream.rs [hunk 3]
```diff
@@ -143,38 +145,40 @@ impl StreamParser {
...
         }
 
-        if self.find_delimiter().is_none() {
-            let frame_text = String::from_utf8_lossy(&self.buffer).trim().to_string();
-            if looks_like_raw_json_frame(&frame_text) {
-                let frame_bytes = std::mem::take(&mut self.buffer);
-                events.extend(self.parse_frame_bytes(frame_bytes)?);
-            }
-        }
+        let _resume_state = (&self.last_event_id, self.retry_ms);
```
[the remark this answers is not in the document]
Comment (aistar-au) [reply src/api/stream.rs]: Addressed in 2f5d86a: the no-op _resume_state read was removed from the parser loop.
